chemprop/nn/loss.py

```diff
@@ -348,3 +349,31 @@ def _calc_unreduced_loss(self, preds: Tensor, targets: Tensor, mask: Tensor, *ar
 
     def extra_repr(self) -> str:
         return f"threshold={self.threshold}"
+
+
+@LossFunctionRegistry.register(["quantile", "pinball"])
+class QuantileLoss(LossFunction):
+    def __init__(self, task_weights: ArrayLike = 1.0, alpha: float = 0.1):
+        super().__init__(task_weights)
+        self.alpha = alpha
+
+        bounds = torch.tensor([-1 / 2, 1 / 2]).view(-1, 1, 1)
+        tau = torch.tensor([[alpha / 2, 1 - alpha / 2], [alpha / 2 - 1, -alpha / 2]]).view(
+            2, 2, 1, 1
+        )
+
+        self.register_buffer("bounds", bounds)
+        self.register_buffer("tau", tau)
```

Comment:
Thanks good!

Reply:
After removing them from the buffer, we have to manually move them to the correct device. Therefore, I create these two objects each time the loss calculation is called. Does that sound okay?

Reply:
Since removing them as buffers would require additional code to handle device movement (https://github.com/chemprop/chemprop/pull/963#discussion_r1787943094), perhaps we should stick with the original design?